src/java.base/share/classes/jdk/internal/classfile/impl/AnnotationReader.java

```diff
@@ -280,4 +281,24 @@ private static int skipTypeAnnotation(ClassReader classReader, int p) {
         p = skipElementValuePairs(classReader, p);
         return p;
     }
+
+    public static void writeAnnotation(BufWriterImpl buf, Annotation annotation) {
+        // handles annotations and type annotations
+        // TODO annotation cleanup later
```

Comment:
Do you have any specific annotation cleanup in mind?

Reply:
I am thinking of removing the Writable Hierarchy from annotations, and these static methods will handle writing with like a switch (on char, can't use pattern match in bootstrap) instead